feat: Optimize bettercap arp spoofing script for dynamic and interactive use

parrth20 opened the pull request:
This PR makes bettercap_arp_spoofing_script.py script more dynamic and interactive and user feirendly , the previous nature of script, which required manual input of IP addresses and fixed delays . this i have replaced with a more intelligent and streamlined workflow.

I'll breakdown changes I have made 

1. Dynamic fullduplex Input :  the script now asks a simple yes/no question, which is much more intuitive.
2. Enhanced Target Selection : this is the biggest improvement  the script now actively helps the user choose their target.

Discovery: It runs net.probe on and waits for the user to press Enter.(no delays as in prev one)
Parsing: It then runs net.show and uses a Regular Expression (regex) to parse the output, extracting the IP, MAC address, and name of each device.
Selection: It displays a clean, numbered list and asks the user to pick targets by number. This allows for selecting a single target or multiple targets easily (e.g., 1,3).

3. Optimized Time Delay : The script now starts discovery and waits for the user to press Enter giving them full control over how long the discovery phase runs. 
4. More modular and cleaner code 

If you need more details about the code you ask me or any further changes.

 Closes https://github.com/amandewatnitrr/hacking-tutorial/issues/5

amandewatnitrr commented:
@parrth20 can you share the screenshot for the working proof, to adhere to the PR guidelines. Also, make sure to have added proper comments within the code.

parrth20 commented:
> @parrth20 can you share the screenshot for the working proof, to adhere to the PR guidelines. Also, make sure to have added proper comments within the code.

Okay I'll add explainatory comments and screenshot.
pls check my Mitm PR also https://github.com/amandewatnitrr/hacking-tutorial/pull/15

parrth20 commented:
> @parrth20 can you share the screenshot for the working proof, to adhere to the PR guidelines. Also, make sure to have added proper comments within the code.

<img width="1440" height="900" alt="image" src="https://github.com/user-attachments/assets/16740fb8-f22b-4c6b-b3fd-13a56c502b4a" />
this is Proof of working code

parrth20 commented:
> @parrth20 can you share the screenshot for the working proof, to adhere to the PR guidelines. Also, make sure to have added proper comments within the code.

i have updated explaintory comments and added a screenshot if want anything else you can tell me , else please tag this PR in hacktoberfest accpeted PR as it's not showing

if you want me to work on any other issue then let me know that issue number

parrth20 commented:
> @parrth20 btw just asking would you like to be a maintainer for the repo post hacktoberfest, just asking. You have been doing quite great work.

Hey @amandewatnitrr, thank you so much for your kind words about my work!

I would love to be a maintainer for this repo. but  since I am focusing on exploring open source and contributing to different projects for Hacktoberfest this month  I'd like to propose starting next month if that works for you. I'll definitely have time to dedicate then.

Btw, I have sent you a connection request on LinkedIn. Looking forward to connecting!!!
